spark3/src/test/java/org/apache/iceberg/spark/source/TestSparkCatalog.java

```diff
@@ -31,7 +30,14 @@
 
   @Override
   public Table loadTable(Identifier ident) throws NoSuchTableException {
-    TestTables.TestTable table = TestTables.load(Spark3Util.identifierToTableIdentifier(ident).toString());
-    return new SparkTable(table, false);
+    String[] parts = ident.name().split("\\$", 2);
+    if (parts.length == 2) {
+      TestTables.TestTable table = TestTables.load(parts[0]);
+      String[] metadataColumns = parts[1].split(",");
+      return new SparkTestTable(table, metadataColumns, false);
```

Comment:
This is just a different way to pass metadata columns in? Is there a better way to test it?

Reply:
I wanted to simply overload `newScanBuilder` in `SparkTestTable` but it did not work. We can get rid of this once we migrate to Spark 3.2 so I don't worry too much.